### ISSUE TITLE
WeeklyPaper 4. 딥러닝과 머신러닝: 포함관계, 하이퍼파라미터 분석

### ISSUE THREAD

**Issue by finneKIM** [finneKIM/WeeklyPaper#5]
<html>
<body>
<h2>1. 딥러닝과 머신러닝 간의 포함관계</h2>
<h3>1-1. 정의</h3>
<ul>
<li>머신러닝(Machine Learning)은 인공지능(AI)의 한 분야로, 사람이 명시적으로 규칙을 코딩하지 않아도 <strong>데이터로부터 학습</strong>하여 패턴을 찾고 예측하는 기술</li>
<li>딥러닝(Deep Learning)은 머신러닝의 하위 개념으로, 인공신경망(Artificial Neural Network)을 기반으로 한 <strong>다층 구조</strong>(Deep Neural Network)를 이용하여 복잡한 데이터 표현을 학습함</li>
</ul>
<h3>1-2. 포함 관계 도식화</h3>
<pre><code>
인공지능 (AI)
└── 머신러닝 (ML)
    └── 딥러닝 (DL)

</code></pre>
<ul>
<li>즉, 딥러닝은 머신러닝에 포함되고, 머신러닝은 인공지능에 포함됨.</li>
<li>딥러닝은 머신러닝의 한 종류이지만, <strong>대규모 데이터셋과 고성능 컴퓨팅 자원을 활용</strong>해 더욱 <strong>복잡하고 정교한 문제 해결</strong>이 가능함</li>
</ul>
<h3>1-3. 비교 표</h3>

구분 | 머신러닝 | 딥러닝
-- | -- | --
정의 | 데이터를 통해 모델을 학습 | 인공신경망을 통해 학습
특징 | 특징을 사람이 설계해야 함 | 특징도 스스로 학습함 (Feature Engineering 불필요)
복잡도 | 비교적 단순한 구조 | 수십~수백 층의 깊은 신경망 구조
성능 | 작은 데이터셋에서 더 우수할 수 있음 | 큰 데이터셋에서 뛰어난 성능 발휘
연산량 | 상대적으로 적음 | GPU 등의 고성능 연산 자원이 필요

</body>
</html>

**Comment by finneKIM** [comment]
## 2. 딥러닝의 성능 향상을 위한 주요 하이퍼파라미터

> 하이퍼파라미터(Hyperparameter): 모델 학습 전 직접 설정해야 하는 값으로, 모델 성능에 큰 영향을 미침.
> 

### 2-1. 학습률(Learning Rate)

- 모델이 가중치를 얼마나 빠르게 업데이트할지를 결정.
- **너무 크면** 발산(unstable), **너무 작으면** 학습 속도가 느려짐.
- 일반적으로 `0.1`, `0.01`, `0.001` 등으로 시도하며, **스케줄링** 기법도 함께 사용.

### 2-2. 배치 크기(Batch Size)

- 한 번에 모델에 공급되는 데이터 샘플 수.
- **작은 배치**: 더 자주 가중치 업데이트, 일반화 성능 좋음
- **큰 배치**: 학습 속도 빠름, 하지만 과적합 위험 있음

### 2-3. 에폭 수(Epochs)

- 전체 데이터셋을 몇 번 반복해서 학습할지 결정.
- 너무 많으면 과적합(overfitting), 너무 적으면 과소적합(underfitting) 가능

### 2-4. 드롭아웃 비율(Dropout Rate)

- 과적합을 방지하기 위한 기법. 학습 중 무작위로 일부 뉴런을 비활성화
- 일반적으로 `0.2 ~ 0.5` 범위에서 설정

### 2-5. 최적화 알고리즘(Optimizer)

- **가중치 조정 방법**을 결정
- 주요 종류:
    - `SGD`: 기본 경사하강법
    - `Adam`: 학습률 자동 조정, 가장 많이 사용됨
    - `RMSprop`, `Adagrad` 등도 있음

### 2-6. 활성화 함수(Activation Function)

- 뉴런이 출력값을 비선형적으로 변환하는 함수
- 종류:
    - `ReLU`: 가장 많이 사용됨
    - `Sigmoid`, `Tanh`: 과거 많이 사용
    - `LeakyReLU`, `ELU`: ReLU의 변형

### 2-7. 네트워크 구조(Architecture)

- 층 수(Depth), 각 층의 뉴런 수(Width), 연결 방식 등
- CNN, RNN, Transformer 등 문제 유형에 따라 구조 선택
